framework/decode/vulkan_stats_consumer.h

```diff
@@ -30,11 +30,14 @@
 #include "generated/generated_vulkan_struct_decoders.h"
 #include "generated/generated_vulkan_consumer.h"
 #include "util/defines.h"
+#include "vulkan/vulkan.hpp"
```

Comment:
Sorry, I didn't catch this on first pass.  This would be the first inclusion of VulkanHpp and I want to be a little more deliberate about adding it.  Can you just use VkExtent2D?

Reply:
@bartosz-muszarski-arm if you haven't done this already I see how it smooths out the feature you added and I'm not in principle against it.  I'm doing a straw poll to see if anyone has any issues with including VulkanHpp.

Reply:
Could we at least reduce this to just including the structs header and not taking in the entire C++ wrapper?

```cpp
#include <vulkan/vulkan_structs.hpp>
```

Reply:
And also turn off every feature we don't need in the headers that has a macro that defaults to enabled. e.g. in our CMake configs for all projects define `VULKAN_HPP_NO_STRUCT_SETTERS`. We can then turn stuff on when we need it, where we need it.